tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_revocation.py

```diff
@@ -0,0 +1,87 @@
+"""
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Description: Load Test client Server Side configuration
+#   Steps :-
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Author: Deepak Punia <dpunia@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copyright Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+import threading
+import argparse
+from pki.client import PKIConnection
+from pki.cert import CertClient
+from timeit import default_timer as timer
+import sys
+import logging
+import pickle
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--hostname", help="<CA hostname>")
+parser.add_argument("--port", help="<CA Port Number>")
+parser.add_argument("--client-cert", help="path for admin.pem certificate")
+parser.add_argument("--number-of-clients", help="Number of thread", type=int)
+parser.add_argument("--number-of-tests-per-client", help="Number of test per thread", type=int)
+parser.add_argument("--cert_sn_file", help="path for certificate serial number file")
+
+log = logging.getLogger()
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+args = parser.parse_args()
+
+# Create a PKIConnection object that stores the details of the CA.
+connection = PKIConnection('https', args.hostname, args.port)
+
+# The pem file used for authentication. Created from a p12 file using the
+# command -
+# openssl pkcs12 -in <p12_file_path> -out /tmp/auth.pem -nodes
+connection.set_authentication_cert(args.client_cert)
+
+# Instantiate the CertClient
+cert_client = CertClient(connection)
+start = timer()
```

Comment:
Loading the serial numbers and splitting the list might take some time. I think we don't want to include that in the measurement either, so I'd move it just before the loop that starts the threads. Same thing for the enrollment test.

Reply:
Updated